src/test/java/uk/gov/pay/adminusers/resources/UserResourceCreateAndGetTest.java

```diff
@@ -249,6 +251,42 @@ public void shouldReturn404_whenGetUser_withNonExistentUsername() throws Excepti
     }
 
     @Test
+    public void shouldReturn200_IfExternalIdsDoExist() throws Exception {
+        String gatewayAccount1 = valueOf(nextInt());
+        String gatewayAccount2 = valueOf(nextInt());
+        Service service = serviceDbFixture(databaseHelper).withGatewayAccountIds(gatewayAccount1, gatewayAccount2).insertService();
+        String serviceExternalId = service.getExternalId();
+        Role role = roleDbFixture(databaseHelper).insertRole();
+        User user = userDbFixture(databaseHelper).withServiceRole(service.getId(), role.getId()).insertUser();
+        User user2 = userDbFixture(databaseHelper).withServiceRole(service.getId(), role.getId()).insertUser();
+        User user3 = userDbFixture(databaseHelper).withServiceRole(service.getId(), role.getId()).insertUser();
+
+        List<String> externalIds = new ArrayList<>();
+        externalIds.add(user.getExternalId());
+        externalIds.add(user2.getExternalId());
+        externalIds.add(user3.getExternalId());
+
+        ImmutableMap<Object, Object> userPayload = ImmutableMap.builder()
+                .put("external_ids", externalIds)
+                .build();
+
+        givenSetup()
+                .when()
+                .body(mapper.writeValueAsString(userPayload))
+                .contentType(JSON)
+                .accept(JSON)
+                .post(USER_EMAILS_RESOURCE_URL)
+                .then()
+                .statusCode(200)
+                .body("results.size()", is(3))
+                .body("results[0].email", notNullValue())
+                .body("results[0].external_id", notNullValue())
+                .body("results[2].email", notNullValue())
+                .body("results[2].external_id", notNullValue());
```

Comment:
I’d prefer if this were a bit more rigourous (i.e. actually check the external IDs and email are correct).

Also, no checks of `results[1]`!